File: frontend/src/api/socratic.ts
import axios from 'axios'

const BASE = import.meta.env.VITE_API_BASE || 'http://localhost:8000'
const http = axios.create({ baseURL: BASE, timeout: 60000 })

http.interceptors.response.use(
  res => res,
  err => {
    const detail = err?.response?.data?.detail
    if (detail) err.message = typeof detail === 'string' ? detail : JSON.stringify(detail)
    return Promise.reject(err)
  }
)

export interface Message {
  role: 'ai' | 'user'
  content: string
}

export interface SessionOut {
  id: number
  trade_id: number
  messages: Message[]
  blind_spots: string[]
  summary: string
  status: 'active' | 'completed'
  created_at: string
  updated_at: string
}

export interface ReplyResponse {
  session_id: number
  ai_message: string
  blind_spots: string[]
  status: 'active' | 'completed'
  summary: string
}

export const socraticApi = {
  start: (trade_id: number) =>
    http.post<SessionOut>('/api/socratic/start', { trade_id }).then(r => r.data),

  reply: (session_id: number, user_message: string) =>
    http.post<ReplyResponse>('/api/socratic/reply', { session_id, user_message })
      .then(r => r.data),

  getByTrade: (trade_id: number) =>
    http.get<SessionOut[]>(`/api/socratic/trade/${trade_id}`).then(r => r.data),

  getSession: (session_id: number) =>
    http.get<SessionOut>(`/api/socratic/session/${session_id}`).then(r => r.data),
}

File: backend/app/schemas/socratic.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional


class Message(BaseModel):
    role: str       # "ai" | "user"
    content: str


class SessionOut(BaseModel):
    id: int
    trade_id: int
    messages: List[Message]
    blind_spots: List[str]
    summary: str
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class StartSessionRequest(BaseModel):
    trade_id: int


class ReplyRequest(BaseModel):
    session_id: int
    user_message: str


class ReplyResponse(BaseModel):
    session_id: int
    ai_message: str
    blind_spots: List[str]
    status: str          # active | completed
    summary: str         # 审讯完成时才有内容
